tests/src/Interop/PInvoke/Miscellaneous/HandleRef/HandleRefTest.cs

```diff
@@ -77,7 +77,8 @@ public unsafe static int Main(string[] args)
             Action gcCallback = () => { Console.WriteLine("GC callback now"); GC.Collect(2, GCCollectionMode.Forced); GC.WaitForPendingFinalizers(); GC.Collect(2, GCCollectionMode.Forced); };
             Assert.AreEqual(intReturn, TestNoGC(hr4, gcCallback), "The return value is wrong");
             Console.WriteLine("Native code finished");
-
+            GC.KeepAlive(boxedInt);
```

Comment:
Why is this needed? The GCHandle that is allocated above (and never freed) should be keeping the `boxedInt` alive.

Reply:
I think there is a different problem here - when ` *int4Ptr = intManaged;` runs, there does not seem to be anything pinning the `BoxedInt` object.

Reply:
Without this, the test was still failing under GCStress in a way different than if the HandleRef wasn't holding the reference. Without this, the value pointed to in the `CollectableClass.PtrToChange` field was changed to `0` during the GC collect passes and GCStress. If the test was failing because of `HandleRef` not holding the reference, the value pointed to should have been changed to `int.MaxValue`.